### PR TITLE
Fixed occasional errors in FileBasedWalTest

### LINKED ISSUE #828: file_based_wal_test scheduling delay
**Describe the bug**
When I running `file_based_wal_test`, it maybe happen an exception:

```
Expected: numFilesAfterGC
Which is: 6
To be equal to: totalFilesNum - expiredFilesNum
Which is: 5
total 9, numFilesAfterGC 6, expiredFilesNum 4
```

This may caused by the `GC` scheduling delay.

### PR DESCRIPTION
**Pull request by critical27**
We need to wait until all ongoing wals has been flushed out, sometimes in CI env, it would failed.

close #828

### CODE REVIEW

**Comment by jude-zhu** [comment]
Jenkins go

**Comment by darionyaphet** [comment]
LGTM !  Maybe we should append some comment to explain the idea next time. Just suggestion!

**Comment by critical27** [comment]
Jenkins go

File: src/kvstore/wal/FileBasedWal.cpp [hunk 1]
```diff
@@ -282,6 +283,10 @@ void FileBasedWal::closeCurrFile() {
     currInfo_->setMTime(::time(nullptr));
     DCHECK_EQ(currInfo_->size(), FileUtils::fileSize(currInfo_->path()))
         << currInfo_->path() << " size does not match";
+    struct utimbuf timebuf;
+    timebuf.modtime = currInfo_->mtime();
```
Comment (darionyaphet) [inline src/kvstore/wal/FileBasedWal.cpp]: make sure it same with file last time

**Comment by critical27** [comment]
For god sake, it finally pass ut.

File: src/kvstore/wal/test/FileBasedWalTest.cpp [hunk 2]
```diff
@@ -362,19 +362,23 @@ TEST(FileBasedWal, TTLTest) {
                            folly::stringPrintf("Test string %02d", i)));
     }
     // Wait for the buffer flushed.
-    sleep(1);
+    while (wal->onGoingBuffersNum_ > 0) {
```
Comment (dangleptr) [inline src/kvstore/wal/test/FileBasedWalTest.cpp]: Do we still need the while scope?  After flusher removed,  onGoingBuffersNum_ will always be zero.

File: src/kvstore/wal/FileBasedWal.cpp (hunk 1 above)
Comment (critical27) [reply src/kvstore/wal/FileBasedWal.cpp]: Yeah, I add a check of return value;

File: src/kvstore/wal/test/FileBasedWalTest.cpp (hunk 2 above)
Comment (critical27) [reply src/kvstore/wal/test/FileBasedWalTest.cpp]: Good suggestion.

File: src/kvstore/wal/FileBasedWal.cpp (hunk 1 above)
Comment (dangleptr) [reply src/kvstore/wal/FileBasedWal.cpp]: Well done!

**Pull request merged**